scripts/miniapps/run_miniapps

```diff
@@ -0,0 +1,264 @@
+PowerOfTwo="
+XSBench
+hpgmg-fv
+"
+
+date
+echo "###"
+echo "### Mini Apps which no special nranks requirements (XSBench, HPGMG)"
+echo "###"
+
+for test in $PowerOfTwo; do
+    if [ $size = "large" ]; then
+        launch_args="$launch_args_large"
+        timeout="$timeout_2"
+    else if [ $size = "medium" ]; then
+        launch_args="$launch_args_medium"
+        timeout="$timeout_1"
+    else
+        launch_args="$launch_args_small"
+        timeout="$timeout_1"
+    fi
+    fi
+    export OMP_NUM_THREADS='2'
+    #echo $OMP_NUM_THREADS
+    if [ $test = "XSBench" ]; then
+        app_args="-s small -t 2 -l 1000"
+    else if [ $test = "hpgmg-fv" ]; then
+        app_args="6 8"
+    fi
+    fi
+    run_test $test "$launch_args" "$timeout" "$app_args"
+done
+
+# For the next group of apps, select any or all of these
+Special="
+snap
+CoMD-openmp-mpi
+lulesh2.0
+"
+#gsnap
+#ksnap
+#isnap
+
+echo "###"
+echo "### Mini Apps with strict launch requirements, varying app parameters etc. (SNAP, LULESH, CoMD)"
+echo "###"
+
+for test in $Special; do
+    #if [ $test = "gsnap" ] || [ $test = "ksnap" ] || [ $test = "isnap" ]; then
+    if [ $test = "snap" ] ; then
+        if [ $size = "large" ]; then
+            launch_args="$launch_args_snap_large"
+            app_args="./8192MT.input ./SNAPlarge.output"
+            export OMP_NUM_THREADS='2'
+            #echo $OMP_NUM_THREADS
+        else if [ $size = "medium" ]; then
+            launch_args="$launch_args_snap_medium"
+            app_args="./96ST4nodes.input ./SNAPmedium.output"
+            export OMP_NUM_THREADS='1'
+            #echo $OMP_NUM_THREADS
+        else
+            launch_args="$launch_args_snap_small"
+            app_args="./6MT.input ./SNAPsmall.output"
+            export OMP_NUM_THREADS='2'
+            #echo $OMP_NUM_THREADS
+        fi
+        fi
+    else if [ $test = "CoMD-openmp-mpi" ]; then
+        if [ $size = "large" ]; then
+            launch_args="$launch_args_large"
+            app_args="-i20 -j20 -k20 -x800 -y800 -z800"
+        else if [ $size = "medium" ]; then
+            launch_args="$launch_args_lulesh_medium"
+            app_args="-i6 -j6 -k6"
+        else
+            launch_args="$launch_args_lulesh_small"
+            app_args="-i2 -j2 -k2"
+        fi
+        fi
+        export OMP_NUM_THREADS='2'
+        #echo $OMP_NUM_THREADS
+    else if [ $test = "lulesh2.0" ]; then
+        if [ $size = "large" ]; then
+            launch_args="$launch_args_large"
+            app_args="-s 48 -i 200"
+            timeout="$timeout_3"
+        else if [ $size = "medium" ]; then
+            launch_args="$launch_args_lulesh_medium"
+            app_args="-s 38 -i 100"
+        else
+            launch_args="$launch_args_lulesh_small"
+            app_args="-s 38 -i 1000"
+        fi
+        fi
+        export OMP_NUM_THREADS='2'
+        #echo $OMP_NUM_THREADS
+    fi
+    fi
+    fi
+    run_test $test "$launch_args" "$timeout" "$app_args"
```

Comment:
Seems like you should probably unset OMP_NUM_THREADS here for tests that may eventually want to use the default.